protostar/utils/compiler/pass_managers.py

```diff
@@ -102,3 +143,61 @@ def visit_CodeElementFunction(self, elm: CodeElementFunction):
         if attr is not None:
             self.add_abi_storage_var_types(elm=attr)
             return
+
+
+class TestCollectorStage(VisitorStage):
```

Comment:
This stage is strictly bound to a particular Visitor. The pattern we did in Cairo compiler was to override `__init__` constructor to forcefully specify the visitor factory and `modify_ast` arguments.

Reply:
You could remove these arguments from `__init__` and provide them directly here. Also you should rather call super() here instead of setting these fields manually